File: legacy/python-backend/app/services/candle/candle_publisher.py
import orjson

from app.cache.redis import redis_client


class CandlePublisher:

    CHANNEL = "candle_events"

    @classmethod
    async def publish(
        cls,
        candle: dict
    ):

        await redis_client.publish(
            cls.CHANNEL,
            orjson.dumps(
                candle
            ).decode()
        )

File: backend/src/services/candlePublisher.ts
import redisClient from "../core/redis";

export class CandlePublisher {
  static CHANNEL = "candle_events";

  static async publish(candle: any) {
    await redisClient.publish(this.CHANNEL, JSON.stringify(candle));
  }
}
export default CandlePublisher;
